### ISSUE TITLE
Conflicting docs around spec.instanceProfile

### ISSUE THREAD

**Issue by acobaugh** [aws/karpenter-provider-aws#5461]
### Description

**How can the docs be improved?**

In the [v1beta1 migration guide](https://karpenter.sh/v0.32/upgrading/v1beta1-migration/#instanceprofile), it states `The Karpenter spec.instanceProfile field has been removed from the EC2NodeClass in favor of the spec.role field.`. However, in the [EC2NodeClass](https://karpenter.sh/v0.33/concepts/nodeclasses/#specinstanceprofile) docs, it clearly claims it still exists. There is also mention of both `spec.role` and `spec.instanceProfile` being available [further on in the same doc](https://karpenter.sh/v0.32/upgrading/v1beta1-migration/#updating-iam-instance-profile-permissions).

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
Yep, you're right that we should update this. Would you be willing to make an update to the doc for a contribution?
